maps-app-ios/UI/Map View/MapViewController+AppPreferences.swift

```diff
@@ -36,4 +34,8 @@ extension MapViewController {
             AppPreferences.viewpoint = changedMapView.currentViewpoint(with: .centerAndScale)
         }
     }
+    
+    func teardownAppPreferences() {
+        navigatingObserver = nil
```

Comment:
Can you `navigatingObserver?.invalidate()` before setting to nil?

Reply:
The willSet does that.

Reply:
of course. Great!

Reply:
(that's what I get for looking only at the PR 6 line snippet of changes)